internal/ingress/controller/controller.go

```diff
@@ -1031,6 +1031,12 @@ func (n *NGINXController) createServers(data []*extensions.Ingress,
 
 			servers[host].SSLCert = *cert
 
+			if n.cfg.DynamicConfigurationEnabled {
+				// Certificates won't be written on disk during dynamic configuration mode
+				servers[host].SSLCert.PemFileName = defaultPemFileName
+				servers[host].SSLCert.PemSHA = defaultPemSHA
+			}
```

Comment:
Could we add some integration tests for this new behaviour? 

A good reference starting point is looking at this file: https://github.com/kubernetes/ingress-nginx/blob/ee46f486c77665096f366eb215aa135032291b5a/test/e2e/lua/dynamic_configuration.go